ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +50,30 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    # Unwrap the handlers
+    _u(starlette.routing.Mount, "handle")
+    _u(starlette.routing.Route, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    def _wrap(scope, receive, send):
+        # Since handle can be called multiple times for one request, we take the path of each instance
+        # Then combine them at the end to get the correct resource name
+        if "__dd_paths__" in scope:
+            scope["__dd_paths__"].append(instance.path)
+
+        else:
+            scope["__dd_paths__"] = [instance.path]
+
+        span = tracer.current_root_span()
```

Comment:
Check that the name value on the span is fastapi.request so that we don't accidentally instrument users custom spans if they made a custom span before ours.

Reply:
may be worth logging something if fastapi.request is not the parent